receiver/factory.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receiver
+
+import (
+	"github.com/census-instrumentation/opencensus-service/processor"
+	"github.com/spf13/viper"
+)
+
+// TraceReceiverFactory is an interface that builds a new TraceReceiver based on
+// some viper.Viper configuration.
+type TraceReceiverFactory interface {
```

Comment:
Given that these interfaces are the same as the factories for processors do we really need them? It can be useful to test if the pipeline is starting with an actual receiver but that seems to come easily by checking that they support some interface specific of receivers.

Reply:
Per offline conversation, you convinced me otherwise :)